simulatingrisk/hawkdovevar/model.py

```diff
@@ -113,3 +114,78 @@ def adjustment_round(self) -> bool:
             and self.schedule.steps > 0
             and self.schedule.steps % self.adjust_round_n == 0
         )
+
+    def get_data_collector_options(self):
+        # in addition to common hawk/dove data points,
+        # we want to include population risk category
+        opts = super().get_data_collector_options()
+        opts["model_reporters"]["population_risk_category"] = "population_risk_category"
+        return opts
+
+    @property
+    def population_risk_category(self):
+        # calculate a category of risk distribution for the population
+        # based on the proportion of agents in different risk categories
+        # (categorization scheme defined by LB)
+
+        # tally the number of agents with each risk level
+        risk_counts = Counter([a.risk_level for a in self.schedule.agents])
+        # count the number of agents in three groups:
+        #  Risk-inclined (RI) : r = 0, 1, 2
+        #  Risk-moderate (RM): r = 3, 4, 5
+        #  Risk-avoidant (RA): r = 6, 7, 8
+        total = {
+            "risk_inclined": risk_counts[0] + risk_counts[1] + risk_counts[2],
+            "risk_moderate": risk_counts[3] + risk_counts[4] + risk_counts[5],
+            "risk_avoidant": risk_counts[6] + risk_counts[7] + risk_counts[8],
+        }
+        # for each group, calculate percent of agents in that category
+        total_agents = len(self.schedule.agents)
+        percent = {key: val / total_agents for key, val in total.items()}
+
+        # majority risk inclined (> 50%)
+        if percent["risk_inclined"] > 0.5:
+            # If < 10% are RM & < 10% are RA: let c = 1
+            if percent["risk_moderate"] < 0.1 and percent["risk_avoidant"] < 0.1:
+                return 1
```

Comment:
Do these integers returned have an integer/quantitative meaning (c is a numeric variable), or do these represent constants or categories? If latter, probably better to use constants pointing to the integers, or even strings

Reply:
this is something that Lara defined as a way to try to categorize the different end states we're seeing on the adaptive risk hawk/dove simulation. I'll share the doc with you on slack so you can help think about it, I'm wondering if there are is a better way to understand & manage it (and it sounds like she wants to add more cases)

Reply:
They represent categories, but the numbers 1-12 are in order from "containing states with the most agents who take the most risks" to "containing states with the fewest agents who take the fewest risks" (though 6 & 7 are incomparable in this respect, since one of them contains a significant number of risk-inclined agents and a significant number of risk-avoidant agents, and the other contains few of each). So they represent the resulting states ordinally, though not cardinally (e.g. higher numbers mean less risk-taking among agents, but 8 doesn't mean, e.g., twice as risk-avoidant as 4).

Reply:
Maybe it's too cute, but what about Roman numerals? They might imply ordinality better, and be a bit more legible as categories at the same time?